xla/tsl/python/lib/core/numpy.h

```diff
@@ -29,6 +29,10 @@ limitations under the License.
 #define NO_IMPORT_ARRAY
 #endif
 
+// Prevent linking error with numpy>=2.1.0
```

Comment:
I'd say a bit more, perhaps:

"Without this define, Numpy's API symbols will have hidden symbol visibility, which may break things if Bazel chooses to build a `cc_library` target into its own `.so` file. Bazel typically does this for debug builds."